code/script.js

```diff
@@ -227,99 +270,298 @@ const setSecret = () => {
 
+  if (category === 'features' || category === 'eyes') {
+    if (value === secret[category]) {
+      keep = true
+    }
+  } else if (category === 'accessories' || category === 'other' || category === 'weapons') {  //these categories are compared differently than features and eyes because these categories contain arrays
+    if (secret[category].includes(value)) {
+      keep = true
+    }
   }
+
+  filterCharacters(keep);
 }
 
+
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
+//shows the alert messages
+  if (category === 'accessories' || category === 'weapons') {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the character has ${value}! Keep all characters that have ${value}.`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the character doesn't have ${value}! Remove all characters that have ${value}.`
       )
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the character is a ${value}! Keep all characters that are ${value}s.`
+      )
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the character is not a ${value}! Remove all characters that are ${value}s.`
+      )
     }
+    } else {
+      if (keep) {
+      alert(
+        `Yes, the character has ${value} ${category}! Keep all characters that have ${value} ${category}.`
+      )
+      } else {
+      alert(
+        `No, the character does not have ${value} ${category}! Remove all characters that have ${value} ${category}.`
+      )
+      }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  // filters by category then puts the new array on the game board - looks like it's removing characters, but it's replacing the original CHARACTERs array (or previous array) with a new one according to filters
+   
```

Comment:
So for this part, I did it in a way that combined the alert messages and the filters, like this:

```
const filterCharacters = (keep) => {
  const { category, value } = currentQuestion

  // Show the correct alert message for different categories
  // Add a filter to keep or remove. Array method .filter
  if (category === 'accessories') {
    if (keep) {
      alert(
        `Yes, the person wears a ${value}! Keep all the people that wears a ${value}`);
        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
    } else {
      alert(
        `No, the person doesn't wear a ${value}! Remove all the people that wears a ${value}`)
        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
    }
  }
```

...etc for the other categories